### DISCUSSION TITLE
Need help with basics

### DISCUSSION THREAD

**Discussion by userHarpreet** [pschatzmann/arduino-audio-tools#1622] [Q&A] [answered]
Hello there, 
I am very new to this, thus very basic questions.

I am trying to interface INMP441 with ESP-WROOM-32 and using my Bluetooth speaker as sink.

Q1: What is the difference between examples basic-i2s-a2dp and stream-i2s-a2dp?
Q2: How can I copy the Left channel to Right as INMP441 gives output to only single channel?
Q3: Can we use a2dp.set_volume to set different volumes to both channels, if not then how it's possible?

**Comment by pschatzmann** [comment, marked as answer]
1: stream-i2s-a2dp shows how to use the Stream API where the basic example just uses the callback provided by the a2dp library: The basic example is more  efficent!

2: There are many possiblilities e.g. 
   - if you leave i2s to provide 2 channels: use a ConverterFillLeftAndRight converter
   - if you set up i2s correctly to provide only 1 channels : use a [ChannelFormatConverterStream](https://github.com/pschatzmann/arduino-audio-tools/wiki/Converting-the-Data-Format)

3: No - you would need to replace the volume handling implemented in the a2dp library with a [VolumeStream](https://github.com/pschatzmann/arduino-audio-tools/wiki/Volume-Control)

ps. dont forget the Wiki about the INMP441, since this is quite tricky to use.

**Reply by userHarpreet** [reply]
Ques: In 2.1 How can we use ConverterFillLeftAndRight  without stream?

**Reply by pschatzmann** [reply]
Just have a look at the [class documentation](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_converter_fill_left_and_right.html)!

Alternatively you can use it together with a ConverterStream as mentioned in the Readme.

**Reply by userHarpreet** [reply]
Thanks a lot for quick responses, I have figured it out.

Great work.

**Reply by userHarpreet** [reply]
Sorry i have a further concern about the Q3, is there support for balance for A2DP, can I achieve this by using that?

**Reply by pschatzmann** [reply]
Did you read the Wiki: there is a chapter about panning!

**Reply by userHarpreet** [reply]
Thanks a lot.
My bad, I was searching for balance instead of panning.
